pkg/chunk/schema_config.go

```diff
@@ -299,15 +299,17 @@ func (cfg *PeriodConfig) dailyBuckets(from, through model.Time, userID string) [
 
 // PeriodicTableConfig is configuration for a set of time-sharded tables.
 type PeriodicTableConfig struct {
-	Prefix string        `yaml:"prefix"`
-	Period time.Duration `yaml:"period,omitempty"`
-	Tags   Tags          `yaml:"tags,omitempty"`
+	Prefix    string        `yaml:"prefix"`
+	Period    time.Duration `yaml:"period,omitempty"`
+	Retention int64         `yaml:"retention,omitempty"`
+	Tags      Tags          `yaml:"tags,omitempty"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *PeriodicTableConfig) RegisterFlags(argPrefix, tablePrefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.Prefix, argPrefix+".prefix", tablePrefix, "DynamoDB table prefix for period tables.")
 	f.DurationVar(&cfg.Period, argPrefix+".period", 7*24*time.Hour, "DynamoDB table period.")
+	f.Int64Var(&cfg.Retention, argPrefix+".retention", 0, "Number of tables to keep, (0 disables retention).")
```

Comment:
Would it be possible to set retention duration?  Having this as a number means I have to know "how long" a table is, which might change.

Reply:
I suppose we could have retention as a duration and get the number of tables to keep by dividing it by the table period. That way you could change the period without needing to update the retention. Is that what you was referring to with that table length may change?

Reply:
Eggsactly

Reply:
I like it, I will fix that

Reply:
It is now a duration, thanks for suggesting that change